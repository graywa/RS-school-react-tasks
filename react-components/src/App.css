.app {
  min-height: 100vh;
  margin: 0 auto;  
}
.container {
  max-width: 1100px;
  width: 100%;
  height: 100%;
  margin: auto;
  padding: 0 25px;
}