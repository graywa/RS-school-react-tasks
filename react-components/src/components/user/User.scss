.user {
  display: flex;
  flex-direction: column;
  gap: 15px;
  max-width: 250px;
  width: 100%;
  padding: 20px;
  border-radius: 10px;
  box-shadow: 0 0 12px 2px #ebebeb;

  &__photo {
    min-height: 150px;
    display: flex;
    justify-content: center;
    align-items: center;
    img {
      margin: auto;
    }
  }

  &__name {
  }

  &__date {
  }

  &__city {
  }

  &__sex {
  }
  span {
    font-weight: 600;
  }
}

