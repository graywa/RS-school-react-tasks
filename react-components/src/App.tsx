import React from 'react';
import { Route, Routes } from 'react-router-dom';
import './App.css';
import Header from './components/header/Header';
import AboutUs from './pages/about-us/AboutUs';
import Forms from './pages/forms/Forms';
import HomePage from './pages/home-page/HomePage';
import Page404 from './pages/page-404/Page404';

class App extends React.Component {
  componentDidMount() {
    localStorage.setItem('searchValue', '');
  }

  render() {
    console.log('render app');
    return (
      <div className="app">
        <Header />
        <div className="container">
          <Routes>
            <Route path="/" element={<HomePage />} />
            <Route path="/about" element={<AboutUs />} />
            <Route path="/forms" element={<Forms />} />
            <Route path="*" element={<Page404 />} />
          </Routes>
        </div>
      </div>
    );
  }
}

export default App;
