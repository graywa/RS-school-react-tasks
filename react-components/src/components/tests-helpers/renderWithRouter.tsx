import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from '../../App';
import AppRouter from '../app-router/AppRouter';

export const renderWidthRouter = (component: JSX.Element, initRoute = '/') => {
  return render(
    <MemoryRouter initialEntries={[initRoute]}>
      {/* <AppRouter /> */}
      {component}
    </MemoryRouter>
  );
};
