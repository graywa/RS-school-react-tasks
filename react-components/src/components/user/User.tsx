import React from 'react';
import './User.scss';

interface IProps {
  key: number;
  name: string;
  date: string;
  city: string;
  sex: string;
  photo: string;
}

class User extends React.Component<IProps> {
  render() {
    const { key, name, date, city, photo, sex } = this.props;
    return (
      <div key={key} className="user">
        <div className="user__photo">
          <img width={150} src={photo} alt="photo" />
        </div>
        <div className="user__name">
          <span>Имя: </span> {name}
        </div>
        <div className="user__date">
          <span>Дата поступления: </span>
          <div>{date}</div>
        </div>
        <div className="user__city">
          <span>Город: </span> {city}
        </div>
        <div className="user__sex">
          <span>Пол: </span> {sex}
        </div>
      </div>
    );
  }
}

export default User;
