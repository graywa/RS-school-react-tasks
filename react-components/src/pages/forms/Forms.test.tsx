import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import App from '../../App';
import { renderWidthRouter } from '../../components/tests-helpers/renderWithRouter';
import User from '../../components/user/User';

describe('render', () => {
  test('render forms page', () => {
    renderWidthRouter(<App />);
    const formsLink = screen.getByText('Forms');
    expect(formsLink).toBeInTheDocument();
    userEvent.click(formsLink);
    const form = screen.getByText('Регистрация');
    expect(form).toBeInTheDocument();
    expect(screen.getByLabelText('Ваше имя:')).toBeInTheDocument();
    expect(screen.getByLabelText('Ваш город:')).toBeInTheDocument();
    expect(screen.getByLabelText('Мужской')).toBeInTheDocument();
    expect(screen.getByLabelText('Выбрать фото')).toBeInTheDocument();
    expect(screen.getByLabelText('Согласен на обработку данных')).toBeInTheDocument();
  });

  test('render user', () => {
    renderWidthRouter(
      <User key={2} name={'Ирина'} city={'Минск'} date={'2021-04-21'} photo={''} sex={'Женский'} />,
      '/forms'
    );

    expect(screen.getByText('Ирина')).toBeInTheDocument();
    expect(screen.getByText('Минск')).toBeInTheDocument();
    expect(screen.getByText('2021-04-21')).toBeInTheDocument();
    expect(screen.getByText('Женский')).toBeInTheDocument();
  });
});

describe('actions in form', () => {
  beforeEach(() => {
    renderWidthRouter(<App />, '/forms');
  });

  test('registration button should be enabled in the beginning', () => {
    const button: HTMLButtonElement = screen.getByRole('button');

    expect(button.disabled).toBe(false);
  });

  test('input name', () => {
    const nameInput = screen.getByLabelText('Ваше имя:');
    userEvent.type(nameInput, 'Валера');
    expect(screen.getByDisplayValue('Валера')).toBeInTheDocument();
    //screen.debug();
  });

  test('choose date', async () => {
    const dateInput: HTMLInputElement = screen.getByLabelText('Дата поступления:');

    await userEvent.type(dateInput, '2000-01-01');

    expect(dateInput.value).toBe('2000-01-01');
  });

  test('choose city', async () => {
    const selectCity = screen.getByRole('combobox');

    await userEvent.selectOptions(selectCity, ['Гомель']);

    expect(screen.getByRole<HTMLOptionElement>('option', { name: 'Гомель' }).selected).toBe(true);
    expect(screen.getByRole<HTMLOptionElement>('option', { name: 'Минск' }).selected).toBe(false);
    expect(screen.getByRole<HTMLOptionElement>('option', { name: 'Брест' }).selected).toBe(false);
  });

  test('choose sex', async () => {
    const radioInput: HTMLInputElement = screen.getByLabelText('Мужской');

    await userEvent.click(radioInput);

    expect(radioInput.checked).toBe(true);
  });

  test('registration button should be enabled, and after sex chosen button should be disabled', async () => {
    const button: HTMLButtonElement = screen.getByRole('button');
    const radioInput: HTMLInputElement = screen.getByLabelText('Мужской');

    expect(button.disabled).toBe(false);

    await userEvent.click(radioInput);

    userEvent.click(button);

    expect(button.disabled).toBe(true);
  });
});
