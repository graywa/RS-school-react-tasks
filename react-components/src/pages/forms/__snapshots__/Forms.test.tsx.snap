// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions in form input name 1`] = `
<input
  id="name"
  name="name"
  type="text"
/>
`;
